DrawMat/Models/Shape.cs

```diff
@@ -92,9 +92,8 @@ public void UpdateBoundingBox()
         BoundingBox groupBox = new BoundingBox();
         foreach (var child in Children)
         {
-            var bbox = child.BBox;
-            groupBox.Include(bbox.MinX - bbox.Margin/2, bbox.MinY - bbox.Margin/2);
-            groupBox.Include(bbox.MaxX + bbox.Margin/2, bbox.MaxY + bbox.Margin/2);
+            groupBox.Include(child.BBox.MinX - child.StrokeThickness/2, child.BBox.MinY - child.StrokeThickness/2);
+            groupBox.Include(child.BBox.MaxX + child.StrokeThickness/2, child.BBox.MaxY + child.StrokeThickness/2);
```

Comment:
That you have to do something more complex here than `groupBox.Include(child.BBox)` should tell you that there's something off and your implementation is suboptimal.